src/lib/y2security/clients/security_policy_proposal.rb

```diff
@@ -109,118 +120,75 @@ def ask_user(param)
       #
       # @return [Array<String>]
       def links
-        main_links = policies.each_with_object([]) do |policy, all|
-          all << action_link("enable", policy.id)
-          all << action_link("disable", policy.id)
-        end
+        links = policies.map { |p| links_builder.links_for_policy(p) } +
+          failing_rules.values.flatten.map { |r| links_builder.links_for_rule(r) }
 
-        main_links + all_issues.each_with_index.map do |issue, idx|
-          if issue.action?
-            action_link("fix", idx)
-          elsif storage_issue?(issue)
-            action_link("storage", idx)
-          end
-        end.compact
+        links.flatten.compact.uniq
       end
 
-      # Parses a link
+      # Returns the warning level
       #
-      # @param link [Array<String, String>] An array containing the the action
-      #   and the id of the element to act on
-      # @see #ask_user
-      def parse_link(link)
-        link.delete_prefix("#{LINK_DIALOG}--").split(":")
-      end
-
-      # Builds a link
+      # Always blocker issues
       #
-      # @param action [String] Action ("enable", "disable" or "fix")
-      # @param id [#to_s] id of the element to act on
-      # @return [String]
-      def action_link(action, id)
-        "#{LINK_DIALOG}--#{action}:#{id}"
+      # @return [Symbol] :blocker
+      def warning_level
+        rules = failing_rules.values.flatten
+
+        return :warning if rules.none? || rules.none?(&:enabled?)
+
+        :blocker
       end
 
-      # Convenience method to get the instance of the policies manager
+      # Returns a warning message when policies issues are found
       #
-      # @return [Y2Security::SecurityPolicies::Manager]
-      def policies_manager
-        Y2Security::SecurityPolicies::Manager.instance
+      # @return [String,nil] Warning message or nil if no issues were found
+      def warning_message
+        rules = failing_rules.values.flatten
+
+        return nil if rules.none?
+
+        _("The system does not comply with the security policy.")
       end
 
-      # All policies
+      # Runs the security policies checks
       #
-      # @return [Array<Y2Security::SecurityPolicies::Policy>]
-      def policies
-        policies_manager.policies
+      # Calling this method updates the list of issues in
+      # Y2Security::Clients::SecurityProposalProposal.issues.
+      #
+      # @see Y2Security::SecurityPolicies::Manager#issues
```

Comment:
the classes referenced above are outdated and we should be mentioning *failing_rules*, right?